model/map/src/main/java/org/keycloak/models/map/storage/MapKeycloakTransaction.java

```diff
@@ -98,17 +100,19 @@ default V getUpdated(V orig) {
      *
      * @param value new version of the value. Must not alter the {@code id} of the entity
      * @param shouldPut predicate to check in commit phase
+     * @return Entity representing the {@code value} in the store. It may or may not be the same instance as {@code value}
      * @see AbstractEntity#getId()
      */
-    void updateIfChanged(V value, Predicate<V> shouldPut);
+    V updateIfChanged(V value, Predicate<V> shouldPut);
 
     /**
      * Instructs this transaction to delete a value associated with the identifier {@code key} from the underlying store
      * on commit.
      *
+     * @return Returns {@code true} if the object has been deleted or result cannot be determined, {@code false} otherwise.
```

Comment:
Could you please provide an example of such situation when the result wouldn't be possible determine? Do I get it right that we might not know the result yet in moment when this method is called?

Reply:
For example upon async deletes (when the caller does is not interested in the result due to some other eventual consistency guarantee)